mcs/class/Mono.Debugger.Soft/Makefile

```diff
@@ -35,6 +32,9 @@ check:
 
 endif
 
+dtest-app.exe: Test/dtest-app.cs $(TEST_HELPERS_SOURCES)
+	$(CSCOMPILE) -r:$(topdir)/class/lib/$(PROFILE)/System.Core.dll -r:$(topdir)/class/lib/$(PROFILE)/System.dll -out:$@ -unsafe $(PLATFORM_DEBUG_FLAGS) -optimize- Test/dtest-app.cs $(TEST_HELPERS_SOURCES)
```

Comment:
1. `TEST_HELPERS_SOURCES ` is only set inside of `ifdef VALID_TEST_PROFILE`
2. You're missing an `-r:$(topdir)/class/lib/$(PROFILE)/mscorlib.dll` here, I suppose this is the reason for the build error.

Reply:
Interesting! That never breaks for me. I've been confused by the breakage, have tried to reproduce on a number of machines from fresh clones, but I haven't seen any dtest-app failures. I can make these requested changes without reproducing them though.